enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,290 @@
+  a few worker nodes to it, then run conformance tests to make sure we don't
+  run into any problems not described in this enhancement.
+
+- Add periodic nightly tests which install a single-node BIP UPI cluster, add
+  worker nodes to it, and check that ingress traffic still works as expected
+  and recovers even after the `router-default` pod gets deleted and
+  rescheduled. Make sure this is still true even after upgrades.
+
+- Add tests on both cloud / UPI that check that a single-node cluster with
+  additional workers recovers after the single control-plane node reboots
+  by running conformance tests post-reboot.
+
+- Add tests on both cloud / UPI that check that a single-node cluster with
+  additional workers recovers after an upgrade by running conformance tests
+  post-upgrade.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+In the non-goals section it's mentioned that this enhancement does not apply to
+clusters which have been installed prior to the enhancement, so their upgrade
+is not discussed.
+
+This enhancement, to the best of my knowledge, should have no problems persisting
+across any type cluster upgrades. The Test Plan section describes how this will be
+tested.
+
+### Version Skew Strategy
+
+Does not apply, to the best of my understanding.
+
+### Operational Aspects of API Extensions
+
+This enhancement does not modify/add any API
+
+#### Failure Modes
+
+This enhancement does not modify/add any API
+
+#### Support Procedures
+
+This enhancement does not modify/add any API
+
+## Implementation History
+
+Not yet applicable
+
+## Drawbacks
+
+- This is another change which would make single-node clusters slightly different from
+  multi-node clusters, and any such difference is naturally not ideal.
+
+- The installer maintainers would have to make sure that the default
+  `IngressController` resource generated for single-node bootstrap-in-place
+  installations is similar enough to the `IngressController` generated by the
+  Openshift Cluster Ingress Operator by default in the absence of an existing
+  resource. i.e. our goal here was to simply change the node selector label
+  from "worker" to "master", but to do that we had to create an entire
+  `IngressController`, one that does not get merged with the default one that
+  would've been created by the OpenShift Cluster Ingress Operator otherwise.
+  Any future important additions to the default `IngressController` created by
+  the OpenShift Cluster Ingress Operator would also have to be added to the
+  installer code where it generated the manifest for single-node
+  bootstrap-in-place installations.
+
+- TBD
+
+## Alternatives
+
+- Implement this change in the OpenShift Cluster Ingress Operator rather than the installer.
```

Comment:
It's already looking at it in order to decide the replica count
https://github.com/openshift/cluster-ingress-operator/pull/519

Reply:
Right, but as mentioned, it's not sufficient to just look at the replica count, we need to look at the platform as well. Because in cloud with a LB or on baremetal with keepalived there's no point in pinning to the master pool.

Anyway, whether the IngerssController change happens in the installer or in the ingress operator is pretty arbitrary, I don't really care either way - I'm mostly interested to discuss whether the idea makes sense in general